src/Bicep.Core/Parser/Parser.cs

```diff
@@ -297,25 +303,112 @@ private SyntaxBase ParenthesizedExpression()
 
         private SyntaxBase FunctionCallOrVariableAccess()
         {
-            var identifier = this.Identifier(b => b.ExpectedFunctionName());
+            if (IsFunctionCall())
+            {
+                return GetFunctionCallSyntax();
+            }
+            // assumes variable call
+            return GetVariableAccessSyntax();
+        }
 
-            if (this.Check(TokenType.LeftParen) == false)
+        /// <summary>
+        /// Method that peeks current and next token, if current token is an identifier and the next is 
+        /// is left parenthesis then assumes a function call
+        /// </summary>
+        private bool IsFunctionCall()
+        {
+            if (reader.Peek().Type == TokenType.Identifier &&
+               reader.Position + 1 <= reader.Count - 1 &&
+               reader.AtPosition(reader.Position + 1).Type == TokenType.LeftParen)
             {
-                // just a reference to a variable, parameter, or output
-                return new VariableAccessSyntax(identifier);
+                return true;
             }
 
+            return false;
+        }
+
+        /// <summary>
+        /// Method that consumes current identifier token and returns VariableAccessSyntax
+        /// </summary>
+        private SyntaxBase GetVariableAccessSyntax()
+        {
+            // TODO: This identifier could be variable, parameter or output, consider a combined error message?
+            var identifier = this.Identifier(b => b.ExpectedVariableIdentifier());
+
+            // just a reference to a variable, parameter, or output
+            return new VariableAccessSyntax(identifier);
+        }
+
+        /// <summary>
+        /// Method that consumes current identifier token and returns PropertyAccessSyntax
+        /// </summary>
+        private SyntaxBase GetPropertyAccessSyntax(SyntaxBase baseExpression, Token dot)
+        {
+            var propertyName = this.Identifier(b => b.ExpectedPropertyIdentifier());
+            return new PropertyAccessSyntax(baseExpression, dot, propertyName);
+        }
+
+        /// <summary>
+        /// Returns a FunctionCallSyntax including all its arguments if they exist
+        /// </summary>
+        private SyntaxBase GetFunctionCallSyntax()
```

Comment:
>Get [](start = 27, length = 3)

nit - We decided not to include the verb as a convention in the parser because almost every method in this class would have to be called `Parse*`.